File: 11_04_2016_Invert_Binary_Tree/chang.cpp
class Solution {
public:
    /**
     * @param root: a TreeNode, the root of the binary tree
     * @return: nothing
     */
    void invertBinaryTree(TreeNode *root) {
        if (root == NULL) {
            return;
        }
        TreeNode *tmp = root->left;
        root->left = root->right;
        root->right = tmp;
        
        invertBinaryTree(root->left);
        invertBinaryTree(root->right);
        
    }
};

File: 11_10_2016_flatten_binarytree_to_linkedlist/chang.py
class Solution:
    # @param root: a TreeNode, the root of the binary tree
    # @return: nothing
    def flatten(self, root):
        # write your code here
        if root == None:
            return
        self.flatten(root.left)
        self.flatten(root.right)
        p = root
        if p.left == None:
            return
        p = p.left
        while p.right:
            p = p.right
        p.right = root.right
        root.right = root.left
        root.left = None
